invenio_vocabularies/contrib/names/services.py

```diff
@@ -36,20 +37,25 @@ def resolve(self, identity, id_, id_type):
 
         # max_records = 1, we assume there cannot be duplicates
         # the loading process needs to make sure of that
-        results = self._read_many(identity, search_query, max_records=1)
+        if many:
+            results = self._read_many(identity, search_query)
+        else:
+            results = self._read_many(identity, search_query, max_records=1)
+        
         # cant use the results_item because it returns dicts intead of records
         total = results.hits.total["value"]
         if total == 0:
             # Not a PID but trated as such
             raise PIDDoesNotExistError(pid_type=id_type, pid_value=id_)
-
-        # (0 < #hits <= max_records) = 1
-        record = self.record_cls.loads(results[0].to_dict())
-        self.require_permission(identity, "read", record=record)
-
-        return self.result_item(
-            self,
-            identity,
-            record,
-            links_tpl=self.links_item_tpl,
-        )
+        if many:
+            return self.result_list(self, identity, results)
```

Comment:
is there permission needed to read many? I had this thought seeing the permission required below

Reply:
In general it doesn't seem that we do a permission check for the read_many and read_all func. But I agree with you we should most likely add it, I am adding it for this one.